tests/retrieveBearerToken.test.js

```diff
@@ -0,0 +1,136 @@
+import { createMocks } from "node-mocks-http";
+import client from "next-auth/client";
+import { retrieve } from "../pages/api/id/[form]/bearer";
+import queryManager from "../lib/integration/queryManager";
+
+jest.mock("next-auth/client");
+
+describe("Test bearer token retrieve API endpoint", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Should return an error 'Malformed API Request' 400 No formID was supplied", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
+    };
+
+    client.getSession.mockReturnValueOnce(mockSession);
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      query: {
+        form: "", //An empty form ID
+      },
+    });
+
+    await retrieve(req, res);
+
+    expect(res.statusCode).toBe(400);
+  });
+
+  it("Should return a statusCode 200 and a Null as token's value associated to a given form (11).", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
+    };
+    client.getSession.mockReturnValueOnce(mockSession);
+    const data = [{ bearer_token: null }];
+    // Mocking query manager
+    jest.spyOn(queryManager, "getResult").mockReturnValue(data);
+    jest.spyOn(queryManager, "executeQuery").mockReturnValue({ rows: [], rowCount: 0 });
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      query: {
+        form: "11",
+      },
+    });
+
+    await retrieve(req, res);
+    const expectedValue = { token: null };
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining(expectedValue));
+    expect(res.statusCode).toBe(200);
+  });
+
+  it("Should return a valid token associated to a form id = 12.", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
+    };
+    client.getSession.mockReturnValueOnce(mockSession);
+    // Query result
+    const data = [{ bearer_token: "toekakdnaodk" }];
+    // Mocking query manager
+    jest.spyOn(queryManager, "getResult").mockReturnValue(data);
+    jest
+      .spyOn(queryManager, "executeQuery")
+      .mockReturnValue({ rows: [{ bearer_token: "toekakdnaodk" }], rowCount: 1 });
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      query: {
+        form: "12",
+      },
+    });
+
+    await retrieve(req, res);
+    const expectedValue = { token: "toekakdnaodk" };
+    expect(res.statusCode).toBe(200);
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining(expectedValue));
```

Comment:
Just use `data[0]` instead of another variable but this is minor

Reply:
There's a  tiny difference between these two variables. `Data` doesn't contain token property and i want to make sure the value is passed to the response as `token: xxxxxxxxx` and not `baerer_token : xxxxx` instead.

Reply:
But yes it can be replaced by the value itself.